javascript-typing-tutor/main.js

```diff
@@ -0,0 +1,19 @@
+var $spanList = document.querySelectorAll('span');
+var current = 0;
+$spanList[current].className += 'underline';
+
+document.addEventListener('keypress', handleKey);
+
+function handleKey(event) {
+  if ($spanList[current].textContent === event.key ||
+    ($spanList[current].textContent === String.fromCharCode(160) &&
```

Comment:
Hmmm, what happens if you just have a regular space character in your HTML instead of `&nbsp;`? If you can just use a space character, I think you could simplify this.

Reply:
If I have just a regular space, it's not putting a line under it.  When I check the elements, the width is 0, so that's why it's not getting an underline.  Using &nbsp, it does have a width and gets underlined.